pytorch_lightning/strategies/horovod.py

```diff
@@ -197,8 +197,10 @@ def _filter_named_parameters(model: nn.Module, optimizer: Optimizer) -> List[Tup
 
     def teardown(self) -> None:
         super().teardown()
-        self._exit_stack.__exit__(None, None, None)
-        self._exit_stack = None
+        # teardown may be called before `_exit_stack` is set
+        if self._exit_stack:
+            self._exit_stack.__exit__(None, None, None)
+            self._exit_stack = None
```

Comment:
this is a bug that would've been caught with typechecking.

horovod is currently on the exemption list: https://github.com/PyTorchLightning/pytorch-lightning/blob/cc43d07db1ab77385feff04c01f040c5cad805a9/pyproject.toml#L71

@speediedan - could you try removing this line and seeing if there are other typing errors that are surfaced? this file looks llike it has typehint coverage already

Reply:
Great point @ananthsub! I see 18 mypy errors (and 2 notes) in `pytorch_lightning/strategies/horovod.py` with the PL mypy config. I'm not actually using Horovod in my PL extension, but am testing for that strategy's use to warn the user (which is my test that picked up the error). I'll def keep an eye out for these in the future but might not be the right person to make the fixes for this file given I'm not fulling testing Horovod.